File: Java/1704_Determine_if_String_Halves_Are_Alike.java
class Solution {
    
    char[] t = {'a','e','i','o','u','A','E','I','O','U'};
    String s;
    public boolean halvesAreAlike(String s) {
        this.s = s;
        
        int a = getCount(0, s.length()/2);
        int b = getCount(s.length()/2, s.length());
        
        return a == b;
    }
    private int getCount(int start, int end){
        int count = 0;
        for(int i = start; i < end; i++){
            char temp = s.charAt(i);
            for(char c : t){
                if(temp == c){
                    count++;
                    break;
                }
            }
        }
        return count;
    }
}

File: CS/1704_Determine_if_String_Halves_Are_Alike.cs
public class Solution {
    public bool HalvesAreAlike(string s) {
        int a = 0;
        int b = 0;
        //HashSet<char> se = new HashSet<char> {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        char[] se = new char[10]  {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        for(int i = 0; i < s.Length/2; i++){
            a += se.Contains(s[i]) ? 1 : 0;
            /*
            if(se.Contains(s[i])){
                a++;
            }
            */
        }
        for(int i = s.Length/2; i < s.Length; i++){
            b += se.Contains(s[i]) ? 1 : 0;
            /*
            if(se.Contains(s[i])){
                b++;
            }
            */
        }
        return a == b;
    }
}
